### PR TITLE
Fixes and tests for metadata labels

### LINKED ISSUE #8737: Search - Getting a 500 error after having applied too many filters
## Observed behavior
When searching I'm getting a 500 error after having applied too many filters. The issue could be caused by the fake data used but should be further investigated.

## Expected behavior
Fully functional search functionality. 

## Steps to reproduce the issue
1. Install the following [build.](https://github.com/learningequality/kolibri/releases/tag/v0.15.0-beta2)
2. Import some channels and run the following management command: `kolibri manage labeltestdata `
3. Filter by as many criteria as possible

## Logs

[logs.zip](https://github.com/learningequality/kolibri/files/7564984/logs.zip)

## Usage Details
 - OS: Windows 10
 - Browser: Chrome

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Adds extensive testing for searching by metadata labels and the available labels generation code
* This showed that the available metadata labels code was not quite doing what it ought
* Changes how the available metadata labels code produces the available metadata labels to make the tests pass and also improve performance
* Uses a Bit wise OR in Postgres to determine which labels are present
* Implements a Bit wise OR shim for SQLite that functions identically to the postgres Bit wise OR, except that the maximum size of bits must be specified in advance
* Uses a bit wise OR to do the label aggregation in a single query without subqueries, which significantly improves the performance, and gives full test compliance
* Limits the data returned for Channels and Languages in the available label metadata to just the ids (reduces query times for these two specific items by roughly 100x for each)
* Bootstraps full language data into the learn page explicitly to compensate for only the ids (to compensate for the full data no longer coming from the available labels)

Fixes #8737

## Reviewer guidance
Does filtering seem to produce the correct remaining labels?
When testing I changed this line to `if True:` https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/core/content/utils/search.py#L91 to repeatably execute the queries that are being optimized here.

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [x] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by pcenov** [comment]
Tested in this [build ](https://buildkite.com/learningequality/kolibri-python-package/builds/4346) and the 500 error is fixed now plus filtering is fully functional.

File: kolibri/core/content/utils/search.py [hunk 1]
```diff
@@ -4,10 +4,17 @@
 """
 import hashlib
 
+try:
+    from django.contrib.postgres.aggregates import BitOr
+except ImportError:
+    BitOr = None
```
Comment (bjester) [inline kolibri/core/content/utils/search.py:10]: Would this ever actually fail to import?

File: kolibri/core/content/utils/search.py [hunk 2]
```diff
@@ -89,27 +97,25 @@ def get_available_metadata_labels(base_queryset, limit_to_known_fields=True):
...
-                    for value in values
-                }
-            )
-            all_values += values
-        result = queryset.values(*all_values).first()
+        base_queryset = base_queryset.order_by()
+        aggregates = {}
+        for field in bitmask_fieldnames:
+            field_agg = field + "_agg"
+            if connections[base_queryset.db].vendor == "sqlite":
```
Comment (bjester) [inline kolibri/core/content/utils/search.py]: With the above `BitOr` potentially being `None`, perhaps this should also include `or BitOr is None`?

**Review by bjester — APPROVED** [review]
Couple comments, but looks good!

File: kolibri/core/content/utils/search.py (hunk 1 above)
Comment (jredrejo) [reply kolibri/core/content/utils/search.py:10]: yes
`ImportError: No module named psycopg2.extras`
in python 2.7

**Review by jredrejo — APPROVED** [review]
I have tested it as much as I have been able (in sqlite only) and it has worked perfectly.
Also, seeing how `pytest kolibri/core/content/test/test_search.py` results in 
`371 passed in 63.65 seconds` it's quite impressive.
Great work!

**Pull request merged**